src/main/java/uk/gov/pay/directdebit/payments/model/ViewPaginationBuilder.java

```diff
@@ -80,10 +84,13 @@ private void buildLinks(long lastPage) {
         lastLink = PaginationLink.ofValue(uriWithParams(searchParams.buildQueryParamString()).toString());
 
         searchParams.withPage(selfPageNum - 1);
-        prevLink = selfPageNum == 1L ? null : PaginationLink.ofValue(uriWithParams(searchParams.buildQueryParamString()).toString());;
+        prevLink = selfPageNum == 1L ? null : 
```

Comment:
there is a fairly amount of logic here and I'd like to see it tested. I know it is implicitly tested elsewhere (like in `PaymentViewServiceTest.java`) but with unit tests here you could check all the paths without having to actually mess with fixtures and so on

Reply:
Added in https://github.com/alphagov/pay-direct-debit-connector/pull/223/commits/c53d425f52adf6d6c6b3b223d5ce71bb99ab5760